web3-react/src/libs/components/Option.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useMemo } from 'react'
+
+import { ConnectionType, getConnection, tryActivateConnector, tryDeactivateConnector } from '../connections'
+
+export const Option = ({
+  isActive,
+  activeConnectionType,
+  connectionType,
+  onActivate,
+  onDeactivate,
+}: {
+  isActive: boolean
+  activeConnectionType: ConnectionType | null
+  connectionType: ConnectionType
+  onActivate: (connectionType: ConnectionType) => void
+  onDeactivate: (connectionType: null) => void
+}) => {
+  const isOptionActive = useMemo(
```

Comment:
nit: I don't think we need to use `useMemo` here. I personally think that this is an example, and we should keep it as simple and minimal as possible. I am generally against early optimizations, and I don't think we ever need to optimize in this code.

(that comment applies to other `useMemo` and `useCallback` in the code too)

This is something I would like to address in a broder context via a design doc, so I am leaving it as a `nit` for now because this approach is currently quite popular throughout our codebase.

Reply:
I like keeping it simple for the example, and as so we can go for it! Would also love to see a document with some guidelines on this. It's true that these optimizations get overused, but there are cases where they are truly valuable. I'm just wondering how you would figure out when to add them - perhaps perf monitoring?